src/generators/NextGenerator.js

```diff
@@ -0,0 +1,232 @@
+
+    // server route configuration
+    const lc = resource.title.toLowerCase();
+    console.log("Paste the following route to your server configuration file:");
+    console.log(
+      chalk.green(`
+server.get('/${lc}/:hash', (req, res) => {
+  return app.render(req, res, '/${lc}', { hash: req.params.hash })
+});
+`)
+    );
+  }
+
+  generate(api, resource, dir) {
+    const lc = resource.title.toLowerCase();
+    const ucf = this.ucFirst(resource.title);
+    const { fields, imports } = this.parseFields(resource);
+
+    const context = {
+      name: resource.name,
+      lc,
+      uc: resource.title.toUpperCase(),
+      ucf,
+      fields,
+      formFields: this.buildFields(resource.writableFields),
+      imports,
+      hydraPrefix: this.hydraPrefix,
+      title: resource.title
+    };
+
+    // Create directories
+    // These directories may already exist
+    [
+      `${dir}/components/common`,
+      `${dir}/config`,
+      `${dir}/error`,
+      `${dir}/interfaces`,
+      `${dir}/pages`,
+      `${dir}/utils`
+    ].forEach(dir => this.createDir(dir, false));
+
+    // copy with patterned name
+    this.createDir(`${dir}/components/${context.lc}`);
+    [
+      // components
+      "components/%s/List.tsx",
+      "components/%s/ListItem.tsx",
+      "components/%s/Show.tsx",
+
+      // pages
+      "pages/%s.tsx",
+      "pages/%ss.tsx"
+    ].forEach(pattern =>
+      this.createFileFromPattern(pattern, dir, context.lc, context)
+    );
+
+    // interface pattern should be camel cased
+    this.createFile(
+      "interfaces/foo.ts",
+      `${dir}/interfaces/${context.ucf}.ts`,
+      context
+    );
+
+    // copy with regular name
+    [
+      // components
+      "components/common/ReferenceLinks.tsx",
+
+      // error
+      "error/SubmissionError.ts",
+
+      // interfaces
+      "interfaces/Collection.ts",
+
+      // utils
+      "utils/dataAccess.ts"
+    ].forEach(file => this.createFile(file, `${dir}/${file}`, context, false));
+
+    // API config
+    this.createEntrypoint(api.entrypoint, `${dir}/config/entrypoint.ts`);
+  }
+
+  getType(field) {
+    if (field.reference) {
+      return field.reference.title;
+    }
+
+    switch (field.range) {
+      case "http://www.w3.org/2001/XMLSchema#integer":
+      case "http://www.w3.org/2001/XMLSchema#decimal":
+        return "number";
+      case "http://www.w3.org/2001/XMLSchema#boolean":
+        return "boolean";
+      case "http://www.w3.org/2001/XMLSchema#date":
+      case "http://www.w3.org/2001/XMLSchema#dateTime":
+      case "http://www.w3.org/2001/XMLSchema#time":
+        return "Date";
+      case "http://www.w3.org/2001/XMLSchema#string":
+        return "string";
+    }
+
+    return "any";
+  }
+
+  getDescription(field) {
+    return field.description ? field.description.replace(/"/g, "'") : "";
+  }
+
+  parseFields(resource) {
+    const fields = {};
+
+    for (let field of resource.writableFields) {
```

Comment:
Could you switch to the functional style? (same for other `for of` loops)

Reply:
Sure, it's a c/cc from Typescript generator. I'll fix this tomorrow